cpp/examples/basic/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+cmake_minimum_required(VERSION 3.18)
+
+project(basic_example VERSION 0.0.1 LANGUAGES C CXX CUDA)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CUDA_ARCHITECTURES "")
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+set(CPM_DOWNLOAD_VERSION 0.27.2) 
+set(CPM_DOWNLOAD_LOCATION "${CMAKE_BINARY_DIR}/cmake/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+
+if(NOT (EXISTS ${CPM_DOWNLOAD_LOCATION}))
+    message(STATUS "Downloading CPM.cmake")
+    file(DOWNLOAD https://github.com/TheLartians/CPM.cmake/releases/download/v${CPM_DOWNLOAD_VERSION}/CPM.cmake ${CPM_DOWNLOAD_LOCATION})
+endif()
+
+include(${CPM_DOWNLOAD_LOCATION})
+
+CPMAddPackage(NAME  cudf
+    GIT_REPOSITORY  https://github.com/rapidsai/cudf
+    GIT_TAG         main
```

Comment:
Hmm, so [this build issue](https://gpuci.gpuopenanalytics.com/job/rapidsai/job/gpuci-v0.20/job/cudf/job/prb/job/cudf-gpu-test/CUDA=11.0,GPU_LABEL=gpu,OS=ubuntu16.04,PYTHON=3.7/1275/console) probably result from a mismatched versioning of dlpack, which was updated in `21.06` at this PR: #8262 . But this example app still attempts to build the `main` (19.02) branch in the CI environment.

The purpose of adding the examples to CI test is to test if any subsequent PRs will break the examples. But currently this only tests whether the CI test builds with the latest release. I can foresee this will bring in a lot of problem at the end of every release (when `main` branch is updated).

I'm exploring a flag/argument for `CMake` that can dynamically build/link to the `current actively developing branch` in CI environment, while keeping it defaulted to `main` outside of CI so that libcudf explorers can still access the stable release. Other ideas are also welcome.

cc @harrism